### PR TITLE
Fix Creating NOOP Tasks on SNAPSHOT Pool

### PR DESCRIPTION
**Pull request by original-brownbear**
Fixing a few spots where NOOP tasks on the snapshot pool were created needlessly.
Especially when it comes to mixed master+data nodes and concurrent snapshots these
hurt delete operation performance needlessly. Also with cloning snapshots incoming and adding potentially more tasks to the `SNAPSHOT` pool it's nice to optimize a little.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -921,7 +935,8 @@ public void cleanup(long repositoryStateId, Version repositoryMetaVersion, Actio
                 // delete would also log a confusing INFO message about "stale blobs".
                 final Set<String> blobNamesToIgnore = deletedSnapshots.stream().flatMap(
                         snapshotId -> Stream.of(GLOBAL_METADATA_FORMAT.blobName(snapshotId.getUUID()),
-                                SNAPSHOT_FORMAT.blobName(snapshotId.getUUID()))).collect(Collectors.toSet());
+                            SNAPSHOT_FORMAT.blobName(snapshotId.getUUID()), INDEX_FILE_PREFIX + previousGeneration))
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:938]: Slightly unrelated but this is all over the cloud logs needlessly (always informs about the expected index-(N-1)) and I figured it's small enough to sneak in here :)

Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:938]: why are we adding `INDEX_FILE_PREFIX + previousGeneration` here?

Comment (fcofdez) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:938]: Ops, I've just seen this https://github.com/elastic/elasticsearch/pull/62152#discussion_r485447600

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/1 (unrelated scroll IT failure)

**Review by fcofdez — APPROVED** [review]
LGTM 👍

**Comment by original-brownbear** [comment]
Thanks Francisco!

**Pull request merged**